CMakeLists.txt

```diff
@@ -80,8 +80,11 @@ endif()
 #
 option(fletch_BUILD_WITH_PYTHON "Build with Python support where appropriate" FALSE)
 if (fletch_BUILD_WITH_PYTHON)
+
   find_package(PythonInterp 2.7 REQUIRED)
   find_package(PythonLibs 2.7 REQUIRED)
+  find_package(NumPy 1.7)
```

Comment:
I did a grep for numpy on all packages and got this 

```
Ceres/
VTK/
OpenCV_contrib/
OpenBLAS/
Caffe/
Binary file Caffe/
Caffe/
OpenCV/
VXL/
Boost/
```

OpenBLAS seems to only use it for benchmark scripts.  

Boost uses it in `Boost/libs/numeric/odeint/performance/performance.py` and some docs and tests. Not sure if we need to require it for Boost though. 

VTK uses it but I believe on of my other PRs disables VTK python (I've been told [probably] that nothing used by fletch uses it)

So that leaves VXL, OpenCV, Caffe, and Ceres. 

A simpler solution might just be to always require it because numpy is easy to install, widely used, and so fundamental to so many packages, especially those that are binding to C.